### ISSUE TITLE
dry_run not working

### ISSUE THREAD

**Issue by hadi-alnehlawi** [sethvargo/gcr-cleaner#44]
Hi Team, 

First of all thanks for this tool, it is helpful.

I was trying the `dry_run` but nothing is written in the `log exporter`, is there any special configuration to set?

 {"repo":"gcr.io/project-name/xxxx", "dry_run":true}

**Comment by sethvargo** [comment]
What version are you running?

**Comment by hadi-alnehlawi** [comment]
Hi @sethvargo , which version are you referring two?

**Comment by sethvargo** [comment]
https://github.com/sethvargo/gcr-cleaner/releases

**Comment by hadi-alnehlawi** [comment]
v0.5.0

**Comment by sethvargo** [comment]
Can you tell me more about how you're running GCR Cleaner? Where are you looking for output?

**Comment by hadi-alnehlawi** [comment]
I am setting it in the body of the scheudle itself, like below
{"repo": "REPO_NAME", "dry_run": "true"}

**Comment by sethvargo** [comment]
What is the response?

**Comment by hadi-alnehlawi** [comment]
Adding dry_run or not is the same, the schedule is actually job is running and clean up the images.

**Comment by sethvargo** [comment]
What is the HTTP response from invoking the service?

**Comment by hadi-alnehlawi** [comment]
{
httpRequest: {
status: 200
}
insertId: "by0jxbg8k0449f"
jsonPayload: {
@type: "type.googleapis.com/google.cloud.scheduler.logging.AttemptFinished"
jobName: "projects/`PROJECTID`/europe-west3/jobs/gcr-cleaner_eu_gcr_io_`PROJECTID`_`REPO`"
targetType: "HTTP"
url: `"SRC_URL`/http"
}
logName: "projects/`PROJECTID`//logs/cloudscheduler.googleapis.com%2Fexecutions"
receiveTimestamp: "2021-08-20T19:25:44.032215785Z"
resource: {
labels: {3}
type: "cloud_scheduler_job"
}
severity: "INFO"
timestamp: "2021-08-20T19:25:44.032215785Z"
}

**Comment by sethvargo** [comment]
That's the cloud scheduler job - what are the logs for the service? Cloud Scheduler doesn't log the response object (which is the other place the response would be)

**Comment by hadi-alnehlawi** [comment]
here is the [logs ](https://ibb.co/Dp44jnN)for the `cloud run`

**Comment by sethvargo** [comment]
Right, so in dry_run mode, those refs aren't actually deleted. That log message is what _would_ happen.

**Comment by hadi-alnehlawi** [comment]
i see, but actually the refs are deleted already

**Comment by sethvargo** [comment]
I'm not sure what you mean?

**Comment by hadi-alnehlawi** [comment]
I meant that as you said in dry_run mode the images in the target repositories must not be deleted, must it?
What happened in dry_run True or False the refs are actually deleted.

**Comment by sethvargo** [comment]
In dry_run mode, the refs are not actually deleted, even though the log message says that.

**Comment by hadi-alnehlawi** [comment]
I am gratefull @sethvargo for your help on this.

Actually, my issue is In dry_run mode, the refs **are** actually deleted.

**Comment by sethvargo** [comment]
Then you must be running an outdated version or you are not passing the request correctly.
